app/js/components/LabResultEntry.jsx

```diff
@@ -371,16 +383,25 @@ const mapStateToProps = (state) => {
     selectedLabConcept,
     openmrs: { CONSTANTS },
     conceptMembers,
+    form,
   } = state;
-  const selector = formValueSelector('result-entry-form');
-  const didNotPerformCheckbox = CONSTANTS && !!(selector(state, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`));
+
+  const formKeyArr = Object.keys(form);
+  let didNotPerformCheckbox;
+  let formKey;
+  if (formKeyArr.length > 0) {
+    const selector = formValueSelector(formKeyArr[0]);
+    didNotPerformCheckbox = CONSTANTS && !!(selector(state, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`));
+    [formKey] = formKeyArr;
+  }
```

Comment:
Can you explain what the above block of code is doing?  It's unclear to me.

Reply:
Yes, I noticed the redux-form generates a random/unique key each time, so I had to first get the key, that is `formKey`.
As demonstrated in the video, when the checkbox is checked the dropdown displays and when it's not checked the dropdown is hidden, I needed to keep track of the state of the checkbox for that, and that is what `didNotPerformCheckbox` is.